pkg/operator/replacement_starter.go

```diff
@@ -0,0 +1,323 @@
+	}
+
+	eventRecorder := events.NewKubeRecorderWithOptions(
+		kubeClient.CoreV1().Events("openshift-authentication-operator"),
+		events.RecommendedClusterSingletonCorrelatorOptions(),
+		componentName,
+		&corev1.ObjectReference{
+			Kind:      "Deployment",
+			Namespace: "openshift-authentication-operator",
+			Name:      "authentication-operator",
+		},
+	)
+
+	return &AuthenticationOperatorInput{
+		kubeClient:                   kubeClient,
+		configClient:                 configClient,
+		operatorClient:               operatorClient,
+		authenticationOperatorClient: authenticationOperatorClient,
+		eventRecorder:                eventRecorder,
+		informers: []SimplifiedInformerFactory{
+			DynamicInformerFactoryAdapter(dynamicInformers), // we don't share the dynamic informers, but we only want to start when requested
+		},
+	}, nil
+}
+
+func CreateControllerInputFromControllerContext(ctx context.Context, controllerContext *controllercmd.ControllerContext) (*AuthenticationOperatorInput, error) {
+	kubeClient, err := kubernetes.NewForConfig(controllerContext.ProtoKubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	configClient, err := configclient.NewForConfig(controllerContext.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	operatorClient, err := operatorclient.NewForConfig(controllerContext.KubeConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	authenticationOperatorClient, dynamicInformers, err := genericoperatorclient.NewClusterScopedOperatorClient(
+		clock.RealClock{},
+		controllerContext.KubeConfig,
+		operatorv1.GroupVersion.WithResource("authentications"),
+		operatorv1.GroupVersion.WithKind("Authentication"),
+		extractOperatorSpec,
+		extractOperatorStatus,
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	eventRecorder := events.NewKubeRecorderWithOptions(
+		kubeClient.CoreV1().Events("openshift-authentication-operator"),
+		events.RecommendedClusterSingletonCorrelatorOptions(),
+		componentName,
+		&corev1.ObjectReference{
+			Kind:      "Deployment",
+			Namespace: "openshift-authentication-operator",
+			Name:      "authentication-operator",
+		},
+	)
+
+	return &AuthenticationOperatorInput{
+		kubeClient:                   kubeClient,
+		configClient:                 configClient,
+		operatorClient:               operatorClient,
+		authenticationOperatorClient: authenticationOperatorClient,
+		eventRecorder:                eventRecorder,
+		informers: []SimplifiedInformerFactory{
+			DynamicInformerFactoryAdapter(dynamicInformers), // we don't share the dynamic informers, but we only want to start when requested
+		},
+	}, nil
+}
+
+func CreateOperatorStarter(ctx context.Context, authOperatorInput *AuthenticationOperatorInput) (OperatorStarter, error) {
+	ret := &authenticationOperatorStarter{
+		eventRecorder: authOperatorInput.eventRecorder,
+		informers:     append([]SimplifiedInformerFactory{}, authOperatorInput.informers...),
+	}
+	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(
```

Comment:
are all these namespaces required? This could be our chance to start small and add only what is needed.

Reply:
> are all these namespaces required? This could be our chance to start small and add only what is needed.

based on my read, I think so